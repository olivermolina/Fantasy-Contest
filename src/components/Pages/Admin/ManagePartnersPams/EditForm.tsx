import React, { useEffect } from 'react';
import {
  FormControlLabel,
  FormHelperText,
  MenuItem,
  Select,
  Switch,
  TextField,
} from '@mui/material';
import Button from '@mui/material/Button';
import { UserStatus, UserType } from '@prisma/client';
import { Controller, useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';
import * as z from 'zod';
import FormControl from '@mui/material/FormControl';
import { ManagePartnersPamsRowModel } from './ManagePartnersPams';
import { NEW_USER_ID } from '~/constants/NewUserId';
import { mapUserTypeLabel } from '~/utils/mapUserTypeLabel';

/**
 * Represents the validation schema for user form fields.
 */
export const EditFormValidationSchema = z.object({
  id: z.string().min(1, { message: 'ID is required' }),
  status: z.boolean().optional(),
  username: z.string().min(1, { message: 'Username is required' }),
  email: z.string().email({
    message: 'Invalid email. Please enter a valid email address',
  }),
  phone: z
    .string()
    .min(8, {
      message: 'Please enter a valid phone number',
    })
    .max(14, {
      message: 'Please enter a valid phone number',
    })
    .optional(),
  type: z.nativeEnum(UserType, {
    errorMap: () => ({ message: 'Please select user type' }),
  }),
  subAdminId: z.string().optional(),
  password: z.string().min(6),
});

/**
 * Defines the form fields for user form.
 */
export type EditFormInputs = z.infer<typeof EditFormValidationSchema>;

/**
 * Represents the properties of `UserForm` component.
 */
interface Props {
  user: ManagePartnersPamsRowModel;
  subAdminUsers: ManagePartnersPamsRowModel[];
  onSubmit: (data: EditFormInputs) => void;
  closeForm: () => void;
}

/**
 * Defines the `UserForm` component.
 */
export default function EditForm(props: Props) {
  const { user, subAdminUsers, closeForm } = props;
  const {
    register,
    handleSubmit,
    formState: { errors },
    watch,
    control,
    reset,
  } = useForm<EditFormInputs>({
    resolver: zodResolver(EditFormValidationSchema),
  });

  /**
   * Resets the form fields when `user` prop changes.
   */
  useEffect(() => {
    reset({
      id: user.id,
      username: user.username || '',
      password: user.id === NEW_USER_ID ? '' : user.username || '',
      email: user.email,
      phone: user.phone?.toString(),
      type: user.type,
      status: user.status === UserStatus.ACTIVE,
    });
  }, [user]);

  return (
    <form onSubmit={handleSubmit(props.onSubmit)}>
      <p className={'bg-selected p-4 text-xl text-white font-bold'}>
        {user.id === NEW_USER_ID ? 'Add ' : 'Edit '} User
      </p>
      <div
        className={
          'flex flex-col  divide-y divide-dashed justify-center shadow-2xl'
        }
      >
        <div className={'grid grid-cols-3 lg:grid-cols-6 p-4 gap-2'}>
          <span className={'font-semibold'}>Active</span>
          <div className={'col-span-2 lg:col-span-5'}>
            <Controller
              name="status"
              control={control}
              defaultValue={false}
              render={({ field }) => (
                <FormControlLabel
                  control={
                    <Switch {...field} checked={field.value} color="primary" />
                  }
                  label=""
                />
              )}
            />
          </div>
        </div>
        <div className={'grid grid-cols-3 lg:grid-cols-6 p-4 gap-2'}>
          <span className={'font-semibold'}>ID</span>
          <div className={'col-span-2 lg:col-span-5'}>
            <TextField
              variant={'outlined'}
              size={'small'}
              fullWidth
              {...register('id')}
              disabled
            />
          </div>
        </div>
        <div className={'grid grid-cols-3 lg:grid-cols-6 p-4 gap-2'}>
          <span className={'font-semibold'}>Email</span>
          <div className={'col-span-2 lg:col-span-5'}>
            <TextField
              variant={'outlined'}
              size={'small'}
              fullWidth
              {...register('email')}
              error={!!errors?.email}
              helperText={errors?.email?.message}
              disabled={user.id !== NEW_USER_ID}
            />
          </div>
        </div>
        <div className={'grid grid-cols-3 lg:grid-cols-6 p-4 gap-2'}>
          <span className={'font-semibold'}>Username</span>
          <div className={'col-span-2 lg:col-span-5'}>
            <TextField
              variant={'outlined'}
              size={'small'}
              fullWidth
              {...register('username')}
              error={!!errors?.username}
              helperText={errors?.username?.message}
              disabled={user.id !== NEW_USER_ID}
            />
          </div>
        </div>

        <div className={'grid grid-cols-3 lg:grid-cols-6 p-4 gap-2'}>
          <span className={'font-semibold'}>Password</span>
          <div className={'col-span-2 lg:col-span-5'}>
            <TextField
              type="password"
              variant={'outlined'}
              size={'small'}
              fullWidth
              {...register('password')}
              error={!!errors?.password}
              helperText={errors?.password?.message}
              disabled={user.id !== NEW_USER_ID}
            />
          </div>
        </div>
        <div className={'grid grid-cols-3 lg:grid-cols-6 p-4 gap-2'}>
          <span className={'font-semibold'}>Phone</span>
          <div className={'col-span-2 lg:col-span-5'}>
            <TextField
              type="number"
              variant={'outlined'}
              size={'small'}
              fullWidth
              {...register('phone')}
              error={!!errors?.phone}
              helperText={errors?.phone?.message}
            />
          </div>
        </div>
        <div className={'grid grid-cols-3 lg:grid-cols-6 p-4 gap-2'}>
          <span className={'font-semibold'}>Type</span>
          <div className={'col-span-2 lg:col-span-5'}>
            <Controller
              name="type"
              control={control}
              defaultValue={user.type}
              render={({ field }) => (
                <FormControl
                  fullWidth
                  error={!!errors?.type}
                  size={'small'}
                  disabled={user.id !== NEW_USER_ID}
                >
                  <Select size={'small'} fullWidth {...field}>
                    <MenuItem key={'empty-status'} value={undefined}>
                      <span className={'italic text-gray-400'}>
                        Select type
                      </span>
                    </MenuItem>
                    {[UserType.SUB_ADMIN, UserType.AGENT].map((userType) => (
                      <MenuItem key={userType} value={userType}>
                        {mapUserTypeLabel(userType)}
                      </MenuItem>
                    ))}
                  </Select>

                  {errors?.type?.message ? (
                    <FormHelperText>{errors?.type?.message}</FormHelperText>
                  ) : null}
                </FormControl>
              )}
            />
          </div>
        </div>
        {watch('type') === UserType.AGENT && (
          <div className={'grid grid-cols-3 lg:grid-cols-6 p-4 gap-2'}>
            <span className={'font-semibold'}>PAM</span>
            <div className={'col-span-2 lg:col-span-5'}>
              <Controller
                name="subAdminId"
                control={control}
                defaultValue={user.subAdminId}
                render={({ field }) => (
                  <FormControl fullWidth error={!!errors?.type} size={'small'}>
                    <Select size={'small'} fullWidth {...field}>
                      <MenuItem key={'empty-status'} value={undefined}>
                        <span className={'italic text-gray-400'}>
                          Select PAM
                        </span>
                      </MenuItem>
                      {subAdminUsers.map((subAdminUser) => (
                        <MenuItem key={subAdminUser.id} value={subAdminUser.id}>
                          {subAdminUser.username}
                        </MenuItem>
                      ))}
                    </Select>
                  </FormControl>
                )}
              />
            </div>
          </div>
        )}
        <div className={'flex flex-row justify-between p-4'}>
          <Button variant={'outlined'} onClick={closeForm}>
            Cancel
          </Button>
          <Button variant={'contained'} type={'submit'}>
            Save
          </Button>
        </div>
      </div>
    </form>
  );
}
