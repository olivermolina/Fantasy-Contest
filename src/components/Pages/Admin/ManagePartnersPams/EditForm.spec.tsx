import { render } from '@testing-library/react';
import EditForm from './EditForm';
import {
  manageUserMocks,
  subAdminUsersMock,
} from './__mocks__/manageUserMocks';

describe('EditForm', () => {
  it('should render correctly', () => {
    const mockProps = {
      user: manageUserMocks[0]!,
      subAdminUsers: subAdminUsersMock,
      closeForm: jest.fn(),
      onSubmit: jest.fn(),
    };

    const { container } = render(<EditForm {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
