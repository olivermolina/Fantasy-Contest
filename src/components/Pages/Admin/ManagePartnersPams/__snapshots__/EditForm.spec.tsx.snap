// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditForm should render correctly 1`] = `
<div>
  <form>
    <p
      class="bg-selected p-4 text-xl text-white font-bold"
    >
      Edit 
       User
    </p>
    <div
      class="flex flex-col  divide-y divide-dashed justify-center shadow-2xl"
    >
      <div
        class="grid grid-cols-3 lg:grid-cols-6 p-4 gap-2"
      >
        <span
          class="font-semibold"
        >
          Active
        </span>
        <div
          class="col-span-2 lg:col-span-5"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
            >
              <span
                class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
              >
                <input
                  class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                  name="status"
                  type="checkbox"
                  value="true"
                />
                <span
                  class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            />
          </label>
        </div>
      </div>
      <div
        class="grid grid-cols-3 lg:grid-cols-6 p-4 gap-2"
      >
        <span
          class="font-semibold"
        >
          ID
        </span>
        <div
          class="col-span-2 lg:col-span-5"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id=":r0:"
                name="id"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-3 lg:grid-cols-6 p-4 gap-2"
      >
        <span
          class="font-semibold"
        >
          Email
        </span>
        <div
          class="col-span-2 lg:col-span-5"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id=":r1:"
                name="email"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-3 lg:grid-cols-6 p-4 gap-2"
      >
        <span
          class="font-semibold"
        >
          Username
        </span>
        <div
          class="col-span-2 lg:col-span-5"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id=":r2:"
                name="username"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-3 lg:grid-cols-6 p-4 gap-2"
      >
        <span
          class="font-semibold"
        >
          Password
        </span>
        <div
          class="col-span-2 lg:col-span-5"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id=":r3:"
                name="password"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-3 lg:grid-cols-6 p-4 gap-2"
      >
        <span
          class="font-semibold"
        >
          Phone
        </span>
        <div
          class="col-span-2 lg:col-span-5"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r4:"
                name="phone"
                type="number"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-3 lg:grid-cols-6 p-4 gap-2"
      >
        <span
          class="font-semibold"
        >
          Type
        </span>
        <div
          class="col-span-2 lg:col-span-5"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-disabled="true"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-type"
                class="MuiSelect-select MuiSelect-outlined Mui-disabled MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-type"
                role="button"
              >
                Partner
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                disabled=""
                name="type"
                tabindex="-1"
                value="AGENT"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined Mui-disabled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-3 lg:grid-cols-6 p-4 gap-2"
      >
        <span
          class="font-semibold"
        >
          PAM
        </span>
        <div
          class="col-span-2 lg:col-span-5"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-subAdminId"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-subAdminId"
                role="button"
                tabindex="0"
              >
                subAdmin1
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="subAdminId"
                tabindex="-1"
                value="subAdmin1"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-3 lg:grid-cols-6 p-4 gap-2"
      >
        <span
          class="font-semibold"
        >
          Timezone
        </span>
        <div
          class="col-span-2 lg:col-span-5"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-timezone"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-timezone"
                role="button"
                tabindex="0"
              >
                America/New_York
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="timezone"
                tabindex="-1"
                value="America/New_York"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-row justify-between p-4"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Cancel
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Save
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
